dapp/index.html

```diff
@@ -1,134 +1,123 @@
+
+        function connect() {
+            client.connect().then((accountAddress) => {
+                currentAccountAddress = accountAddress;
+                document.getElementById('accountAddress').innerHTML = currentAccountAddress;
+            });
+        }
+
+        function vote() {
+            const contractIndex = document.getElementById('votingContractIndex').value;
+            const votingOption = document.getElementById('votingOption').value;
+
+            client
+                .sendTransaction(
                     currentAccountAddress,
-                    concordiumSDK.AccountTransactionType.InitContract,
+                    concordiumSDK.AccountTransactionType.Update,
                     {
                         amount: new concordiumSDK.CcdAmount(0n),
-                        moduleRef: moduleReference,
-                        initName: "voting",
+                        contractAddress: {
+                            index: contractIndex,
+                            subindex: 0n,
+                        },
+                        receiveName: 'voting.vote',
                         maxContractExecutionEnergy: 3000n,
                     },
-                    {
+                    votingOption,
+                    rawModuleSchema,
+                )
+                .then((msg) => alert(`Successfully sent vote with transaction hash: "${msg}"`))
+                .catch(alert);
+        }
+
+        function idCheckedVote() {
+            const contractIndex = document.getElementById('votingContractIndex').value;
+            const votingOption = document.getElementById('votingOption').value;
+
+            const statement = new concordiumSDK.IdStatementBuilder()
+                .addNonMembership(concordiumSDK.AttributesKeys.countryOfResidence, [votingOption])
+                .getStatement();
+            // A challenge is some random value included in the proof so they cannot be reused.
+            // In a real setup we would verify that the proof is valid and includes the challenge.
+            // And we would check that the included challenge is not older than x minutes.
+            const challenge = 'AAAAAAAA';
+            if (currentAccountAddress != null) {
+                client
+                    .requestIdProof(currentAccountAddress, statement, challenge)
+                    .then((proof) => {
+                        console.log("Got valid ID proof:", proof);
+                        vote();
+                    })
+                    .catch((error) => {
+                        console.log("Could not create ID proof:", error);
+                        alert("You cannot vote on your own country! We busted a cheater!");
+                    });
+            } else {
+                alert("You need to connect first.");
+            }
+        }
+
+        function setupElection() {
```

Comment:
I think SetupElection should be placed between `connect` and `vote`, so it matches the order you implement it in.